src/hunspell/hunspell.cxx

```diff
@@ -1506,7 +1529,18 @@ std::vector<std::string> HunspellImpl::analyze(const std::string& word) {
       else
         reverseword(result);
     }
-    return line_tok(result, MSEP_REC);
+    slst = line_tok(result, MSEP_REC);
+    // output conversion
+    rl = (pAMgr) ? pAMgr->get_oconvtable() : NULL;
```

Comment:
These snippets seem repetitive. Could you extract common code to a helper function?

Please also consider adding a test which will demonstrate the issue, and prevent regressions.